node/pkg/txverifier/evm.go

```diff
@@ -181,12 +181,101 @@ func (tv *TransferVerifier[ethClient, Connector]) pruneCache() {
 	}
 
 	if numPrunedEvals > 0 {
-		tv.logger.Debug("pruned cached receipt evaluations",
+		tv.logger.Info("pruned receipt evaluations cache",
 			zap.Int("numPruned", numPrunedEvals),
 			zap.Int("cacheSize", len(tv.evaluations)),
 		)
 	}
-	// TODO: The other caches should be pruned here too.
+
+	// For the other caches, just prune them if they're too big.
+	if len(tv.chainIdCache) >= CacheMaxSize {
```

Comment:
Worth making this a helper function instead of copying multiple times for each cache?

Reply:
It gets a little annoying because the keys have different types (string vs common.Address) but I could do this

Reply:
Would consider this a nice to have if you're squeezed for time